tfcoreml/_layers.py

```diff
@@ -955,6 +955,10 @@ def argmax(op, context):
     assert False, 'ArgMax: Axis tensor not found in the list of Consts'
   if len(input_shape) == 4 and axis_tf == 3:
     axis = 'C'
+  elif (len(input_shape) == 4 or len(input_shape) == 3) and axis_tf == 2:
+    axis = 'W'
+  elif (len(input_shape) == 4 or len(input_shape) == 3 or len(input_shape) == 2) and axis_tf == 1:
+    axis = 'H'
```

Comment:
This is not entirely correct in general. It tackles issue #177, but may cause other models to convert incorrectly. 
Unfortunately, due to the difference in how TF and CoreML deal with multi-dimensional arrays, its tricky to implement a correct general strategy and so far I haven't been able to come up with one that works in all cases. 

Here is the logic used through the tfcoreml converter to map TF tensors to static CoreML shapes: 
- rank 1 TF tensor, interpreted as [C] for CoreML
- rank 2 TF tensor, interpreted as [B,C] for CoreML 
        (this is ambiguous as it could also mean [H,W] or [W,C] or [H,C] etc)
- rank 3 TF tensor, interpreted as [H,W,C] for CoreML 
       (this is ambiguous as it could also mean [B,H,W] etc)
- rank 4 TF tensor, interpreted as [B,H,W,C] for CoreML